projects/components/src/menu-dropdown/menu-dropdown.component.ts

```diff
@@ -0,0 +1,46 @@
+import { ChangeDetectionStrategy, Component, ContentChildren, Input, QueryList } from '@angular/core';
+import { IconType } from '@hypertrace/assets-library';
+import { IconSize } from '../icon/icon-size';
+import { MenuItemComponent } from './menu-item.component';
+
+@Component({
+  selector: 'ht-menu-dropdown',
+  styleUrls: ['./menu-dropdown.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  template: `
+    <div class="menu-dropdown">
+      <ht-popover [closeOnClick]="true" class="menu-container">
+        <ht-popover-trigger>
+          <div class="trigger-content">
+            <ht-label *ngIf="this.label" class="trigger-label" [label]="this.label"> </ht-label>
+            <ht-icon *ngIf="this.icon" class="trigger-icon" [icon]="this.icon" size="${IconSize.Small}"></ht-icon>
+          </div>
+        </ht-popover-trigger>
+        <ht-popover-content>
+          <div class="dropdown-content">
+            <div *ngFor="let item of items" (click)="item.action()" class="menu-item">
```

Comment:
What happens if action is undefined?

Reply:
as of now, nothing happens. the consumer needs to handle the actions. We could log a warning.

Reply:
So no console error on click for an undefined method? If no error, this is fine. Not looking to log a warning, just didn't want an undefined method to not be handled.